src/zombies.test.ts

```diff
@@ -1,30 +1,121 @@
-import { ok } from "node:assert/strict";
+import { ok, equal } from "node:assert/strict";
 import { test } from "node:test";
 
+const numIsValidInteger = (num: unknown): num is number => {
+  return typeof num === 'number' && Number.isInteger(num) && num >= 0;
+};
+
+const isValidString = (name: string): name is string => {
+  return typeof name === 'string' && name.length > 0;
+};
+
+type ZombieReturnObj = {
+  isFull: () => boolean;
+  addZombie: (zombie: string) => void;
+  zombiesInRoom: () => string[];
+  noZombiesInRoom: () => number;
+  spaceLeft: () => number; 
+};
+
 const createRoom = (capacity: number) => {
-  const _capacity = capacity;
+  if (!numIsValidInteger(capacity)) return `Error: Parameter 'capacity' must be a positive integer`;
+  
+  const _capacity = capacity; 
+  let _zombiesInRoom: string[] = [];
 
   return {
-    isFull: () => true,
+    isFull: () => {
+      if (_capacity === 0 || _capacity === _zombiesInRoom.length) return true;      
+      
+      return false;
+    },
+    addZombie: (zombie: string) => {
+      if (isValidString(zombie)) {
+        if (_capacity === _zombiesInRoom.length) {
+          _zombiesInRoom.splice(0, 1, zombie);
+        } else {
+          _zombiesInRoom.push(zombie);
+        }  
+      } else {
+        console.log(`*** No zombie was added to the room since you tried to add "${zombie}". ***`);
+      }          
+    },
+    zombiesInRoom: () => _zombiesInRoom,
+    noZombiesInRoom: () => _zombiesInRoom.length, 
+    spaceLeft: () => capacity - _zombiesInRoom.length,
   };
 };
 
-test("room is not full", () => {
-  const room = createRoom(0);
+test("room is full", () => {
+  const room = createRoom(0) as ZombieReturnObj;
```

Comment:
There's no reason to add the type here. If you really wanted to do it, then a better solution would be to write:
```typescript
const room: Room = createRoom(0);
```